pkgs/by-name/sq/squawk-chat/package.nix

```diff
@@ -0,0 +1,54 @@
+{ lib
+, stdenv
+, fetchgit
+, cmake
+, kconf
+, lmdb
+, qxmpp
+, boost
+, libomemo-c
+, pkg-config
+, libsForQt5
+, imagemagick
+
+, kwalletSupport ? false # secure password storage
+, kioSupport ? false # better show in folder action
+, kconfSupport ? false # system themes support
+}:
+
+stdenv.mkDerivation rec {
+  pname = "squawk-chat";
+  version = "0.2.3";
+
+  src = (fetchgit {
+    url = "https://git.macaw.me/blue/squawk";
+    rev = "fb843a13463e6600f86d9dde5aa728cdcfa57259";
+    hash = "sha256-dPeB2DOJ22shgeV6UBXb6Kcswre4q3CRaWogLsLyl2U=";
+  }).overrideAttrs (_: {
+    GIT_CONFIG_COUNT = 1;
+    GIT_CONFIG_KEY_0 = "url.https://git.macaw.me/.insteadOf";
+    GIT_CONFIG_VALUE_0 = "gitea@git.macaw.me:";
+  });
+
+  nativeBuildInputs = [ cmake boost libomemo-c pkg-config libsForQt5.wrapQtAppsHook imagemagick libsForQt5.kwallet libsForQt5.kio ];
+  buildInputs = [ libsForQt5.qt5.qtbase lmdb qxmpp ]
+    ++ lib.optional kwalletSupport libsForQt5.kwallet
+    ++ lib.optional kioSupport libsForQt5.kio
+    ++ lib.optional kconfSupport kconf;
```

Comment:
That still does not make sense to me, but i've never worked with lib.optional. [NixPkgs wiki dependencies](https://nixos.org/manual/nixpkgs/stable/#ssec-stdenv-dependencies-overview)

`lib.optional` could be better explained in docs, but there are some examples on github [1](https://github.com/NixOS/nixpkgs/blob/7e1eefd42612868b2f01a22b0768910dbbf8eeb7/pkgs/tools/security/sirikali/default.nix#L47) [2](https://github.com/NixOS/nixpkgs/blob/7e1eefd42612868b2f01a22b0768910dbbf8eeb7/pkgs/applications/networking/remote/remmina/default.nix#L48) which show how it is used in such scenario.

I think more experienced dev than me should review this.

Reply:
I have switched from `lib.optional` to `lib.optionals` as shown in those other repos